play-json/js/src/main/scala/StaticBinding.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <https://www.lightbend.com>
+ */
+package play.api.libs.json
+
+import java.io.InputStreamReader
+
+import scalajs.js, js.JSON
+
+object StaticBinding {
+  /** Parses a [[JsValue]] from raw data (assuming UTF-8). */
+  def parseJsValue(data: Array[Byte]): JsValue =
+    parseJsValue(new String(data, "UTF-8"))
+
+  /** Parses a [[JsValue]] from a stream (assuming UTF-8). */
+  def parseJsValue(stream: java.io.InputStream): JsValue = {
+    var in: InputStreamReader = null
+
+    try {
+      in = new java.io.InputStreamReader(stream, "UTF-8")
+      val acc = StringBuilder.newBuilder
+      val buf = Array.ofDim[Char](1024)
+
+      @annotation.tailrec
+      def read(): String = {
+        val r = in.read(buf, 0, 1024)
+
+        if (r == 1024) {
+          acc ++= buf
+          read()
+        } else if (r > 0) {
+          acc ++= buf.slice(0, r)
+          read()
+        } else acc.result()
+      }
+
+      parseJsValue(read())
+    } catch {
+      case err: Throwable => throw err
+    } finally {
+      if (in != null) in.close()
+    }
+  }
+
+  /** Parses a [[JsValue]] from a string content. */
+  def parseJsValue(input: String): JsValue =
+    anyToJsValue(JSON parse input)
+
+  def generateFromJsValue(jsValue: JsValue, escapeNonASCII: Boolean): String =
+    fromJs(jsValue, escapeNonASCII, 0, _ => "")
+
+  private def fromJs(jsValue: JsValue, escapeNonASCII: Boolean, ilevel: Int, indent: Int => String, newline: Boolean = false, fieldValueSep: String = ":", arraySep: (String, String, String) = ("[", ",", "]")): String = {
```

Comment:
ScalaJS `JSON` doesn't provide an appropriate pretty-printer, so implement one compatible with the existing tests/specs